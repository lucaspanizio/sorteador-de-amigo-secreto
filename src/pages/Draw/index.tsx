export const Draw = () => {
  return <>Sorteio</>;
};
