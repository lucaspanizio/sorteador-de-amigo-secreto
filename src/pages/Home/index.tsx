import { Header } from '@/components/Header';
import { Form } from '@/components/Form';
import { AttendeesList } from '@/components/AttendeesList';

export const Home: React.FC = () => {
  return (
    <>
      <Header />
      <Form />
      <AttendeesList />
    </>
  );
};
