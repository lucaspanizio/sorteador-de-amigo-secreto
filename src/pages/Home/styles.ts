import styled from 'styled-components';

export const Footer = styled.footer`
  display: flex;
  align-items: center;
  justify-content: space-between;

  @media screen and (max-width: 800px) {
    flex-direction: column;

    img {
      margin-top: 32px;
    }
  }
`;
