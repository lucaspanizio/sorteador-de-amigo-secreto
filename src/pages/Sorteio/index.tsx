export const Sorteio = () => {
  return <>Sorteio</>;
};
