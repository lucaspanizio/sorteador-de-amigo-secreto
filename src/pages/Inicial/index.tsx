import { Form } from '@/components/Form';
import { AttendeesList } from '@/components/AttendeesList';
import { Button } from '@/components/Button';
import * as S from './styles';

export const Inicial: React.FC = () => {
  return (
    <section>
      <h2>Vamos come√ßar!</h2>
      <Form />
      <AttendeesList />
      <S.Footer>
        <Button />
        <img src="/images/sacolas.png" alt="Sacolas de compras" />
      </S.Footer>
    </section>
  );
};
