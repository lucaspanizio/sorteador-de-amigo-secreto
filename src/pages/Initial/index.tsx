import { Form } from '@/components/Form';
import { AttendeesList } from '@/components/AttendeesList';
import { useAttendees } from '@/hooks/useAttendees';
import { useNavigate } from 'react-router-dom';
import { useDrawer } from '@/hooks/useDrawer';
import * as S from './styles';

export const Initial = () => {
  const { attendees } = useAttendees();
  const navigate = useNavigate();
  const draw = useDrawer();

  const startGame = () => {
    draw();
    navigate('/sorteio');
  };

  return (
    <section>
      <h1>Vamos come√ßar!</h1>
      <Form />
      <AttendeesList />
      <S.Footer>
        <S.Button
          data-testid="btnIniciar"
          disabled={attendees.length < 3}
          onClick={startGame}
        >
          Iniciar brincadeira!
        </S.Button>
        <img src="/images/sacolas.png" alt="Sacolas de compras" width="150px" />
      </S.Footer>
    </section>
  );
};
