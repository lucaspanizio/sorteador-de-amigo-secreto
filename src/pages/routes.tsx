import { createBrowserRouter, Navigate } from 'react-router-dom';
import { Layout } from '@/components/Layout';
import { Initial } from '@/pages/Initial';
import { Sorteio } from '@/pages/Sorteio';

export const router = createBrowserRouter([
  {
    path: '/',
    element: <Layout />,
    children: [
      { path: '', element: <Initial /> },
      { path: 'sorteio', element: <Sorteio /> },
    ],
  },
]);
