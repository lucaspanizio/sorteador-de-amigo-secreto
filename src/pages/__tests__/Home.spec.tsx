import { RecoilRoot } from 'recoil';
import { render } from '@testing-library/react';
import { Home } from '@/pages/Home';

const mockNavigation = jest.fn();

jest.mock('react-router-dom', () => {
  return {
    useNavigate: () => mockNavigation,
  };
});

describe('Home', () => {
  it('Shoud be render', () => {
    const { container } = render(
      <RecoilRoot>
        <Home />
      </RecoilRoot>,
    );

    expect(container).toMatchSnapshot();
  });
});
