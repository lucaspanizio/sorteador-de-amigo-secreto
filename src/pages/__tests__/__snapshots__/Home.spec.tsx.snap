// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Shoud be render 1`] = `
<div>
  <section>
    <h2>
      Vamos come√ßar!
    </h2>
    <form>
      <input
        class="sc-blHHSb jBMeZG"
        placeholder="Insira os nomes dos participantes"
        type="text"
        value=""
      />
      <button
        class="sc-gtLWhw kthIhm"
        disabled=""
        type="submit"
      >
        Adicionar
      </button>
    </form>
    <ul />
    <footer
      class="sc-dntaoT doFTcL"
    >
      <button
        class="sc-fAUdSK jodoHX"
        disabled=""
      >
        Iniciar brincadeira!
      </button>
      <img
        alt="Sacolas de compras"
        src="/images/sacolas.png"
      />
    </footer>
  </section>
</div>
`;
